### ISSUE TITLE
Callback's Reply Markup Unique is in its data field

### ISSUE THREAD

**Issue by matinhimself** [tucnak/telebot#369]
First of all, I think this issue should be raised so far, So excuse me if it's duplicated and I couldn't find it.

While handling a (button) callback the `CallBack` button's `unique` field is empty, But `data` field has both data and unique fields.

with
```Golang
btnDlt := (&tb.ReplyMarkup{}).Data("Delete🗑", "delete")

app.Handle(&btnDlt,  func(c *tb.Callback) {
	fmt.Printf("%+v", c.Message.ReplyMarkup.InlineKeyboard)
})
```
we will have
`[[{Unique: Text:Delete URL: Data:\fdelete|id InlineQuery: InlineQueryChat: Login:<nil>}]]`

**Comment by rusq** [comment]
I'm not a dev, but spent a bit of time examining the code chasing rabbits, so will be pro-active and answer your Q.

The callback_data is used by the library internally to do the handling of the markup items, normally, you should not be using it unless you really know what you're doing.

Please see the explanation on the simple example below:

Buttons are defined  as:
```go
// ... 
// Function signature is: markup.Data(Label, Unique, Data...)
btn1 := markup.Data("English (US)", "f20315fe6861e1371437f94ce5be9736efbc4fed", "English (US)")
btn2 := markup.Data("Русский", "5537fc8e69d7e9830f739eb94071c2212b4027ad", "Русский")
//...
```
Internally, sha1-hash1 and sha1-hash2 will be used as a key in internal `handlers` map to create handlers for each of the buttons, then this will be concatenated with the data and assigned to "callback_data" field in the JSON that is sent to API.

When you receive the response, it will naturally be there.

I.e.
```json
// ...
"reply_markup": {
    "inline_keyboard": [
      [
        {
          "text": "English (US)",
          "callback_data": "\u000cf20315fe6861e1371437f94ce5be9736efbc4fed|English (US)",
          "switch_inline_query_current_chat": ""
        },
        {
          "text": "Русский",
          "callback_data": "\u000c5537fc8e69d7e9830f739eb94071c2212b4027ad|Русский",
          "switch_inline_query_current_chat": ""
        }
      ]
    ]
  }
//...
```

So, what will happen when library handles it?

[this is simplified algo] 

1. the string `"\u000c5537fc8e69d7e9830f739eb94071c2212b4027ad|Русский"` is matched against the Regexp expression to verify if it's a valid callback data that can be processed by the library.
2. It is then split into `5537fc8e69d7e9830f739eb94071c2212b4027ad` and `Русский`, with the former being used as a Key to find the `handler`, and the latter is assigned to `telebot.Callback.Data` member variable
3. The `handler` is called with parsed data.

SO: **Normally, you access the `Callback.Data` field to get the data you have assigned.**

```go
btnDlt := (&tb.ReplyMarkup{}).Data("Delete🗑", "delete", "<here comes you data to be handled in callback>")

app.Handle(&btnDlt,  func(c *tb.Callback) {
	fmt.Printf("%+v", c.Message.ReplyMarkup.InlineKeyboard)
	fmt.Printf("%+v", c.Data)
})
```

As always, I can't recommend https://github.com/go-delve/delve enough, it really helps to step through the code to see what's happening.

Hope that answers your question.

**Comment by matinhimself** [comment]
Thank you for your _complete_ answer and your advice. 

The issue was about accessing to `unique` field of the **All** buttons of `CallBack.Message.ReplyMarkup.InlineKeyboard` items, Not just `CallBack` And I think I couldn't express it well.

What I was trying to do was making a Dynamic CallBack handler for a button that can be toggled on and off, and I was using it in many different messages with different reply markups, So I needed to rebuild the message with same buttons but different toggle button, then I found out a bug in my app that was created because of `unique` field that I was using to identify toggle button from others was empty. I fixed it by splitting data but I thought it could be Bug in library.
